src/header-utilities/exponentialSearch.hpp

```diff
@@ -0,0 +1,51 @@
+class ExponentialSearch {
+  template< typename RandomAccessIterator,
+	    typename Value,
+	    typename Compare >
+  static RandomAccessIterator apply( RandomAccessIterator left,
+				     RandomAccessIterator right,
+				     RandomAccessIterator end,
+				     Value&& value,
+				     Compare&& compare ){
+    if ( left == right ){ return right; }
+    const auto last = std::prev(right);
+    return ( not compare( *last, value ) ) ?
```

Comment:
This is very confusing about what it does; it's too terse to be easily read. I'm not suggesting it is wrong, I just can't tell what it is doing.